stats-test.cpp

```diff
@@ -20,4 +21,7 @@ TEST_CASE("average is NaN for empty array") {
     
     //Design the REQUIRE statement here.
     //Use http://www.cplusplus.com/reference/cmath/isnan/
+    REQUIRE(isnan(computedStats.average) == true);
+    REQUIRE(isnan(computedStats.max) == true);
+    REQUIRE(isnan(computedStats.min) == true);
```

Comment:
This test is correct. Can you think of other tests? What about a NaN in the input? Write a test and a REQUIRE first, to determine what needs to happen. Then test and fix if necessary.

Reply:
In case of max and min element for vector having atleast one NAN is not a problem. What should I consider for average? Should the average be NAN or average of non NAN numbers?

Reply:
Great question! 👍 
Part of the design of the test case is to reason out what real-world behavior would be expected. Normally statistics are used to aggregate some measurements. What are some of the situations where statistics are needed, and why would some measurements be invalid/NaN?

Reply:
Good reasoning 👍